src/ui/public/management/section.js

```diff
@@ -0,0 +1,68 @@
+import {assign} from 'lodash';
+import IndexedArray from 'ui/indexed_array';
+
+export default class ManagementSection {
```

Comment:
Could this be a factory function rather than a class? It might be helpful in the future if we need to allow plugins to call register for a section in arbitrary order (as discussed above). Then the initialized sections could be tracked by this module rather than outside of it. Just a thought.